internal/metrics/metrics.go

```diff
@@ -131,6 +141,17 @@ func (m Metrics) Collect() ([]byte, error) {
 	r.Partitions = m.getPartitions()
 	r.Screens = m.getScreens()
 
+	// if we're on an architecture with hwcap then find which version
```

Comment:
The one thing I'm not fully understanding about the changes you're requesting is how to handle the chicken/egg problem of needing the architecture in order to do a direct assignment of m.hwCap, but needing m to already be instantiated in order to find the architecture. I'll give it a shot with a direct assignment of m.hwCap to "nil" and override if necessary in the Collect() function.

Reply:
Said differently, I think you can keep ldPath and libc6Cmd as local variables in the constructor (don’t set it to Metrics), and when you create m, directly assign m.hwCapCmd based on those values.

Reply:
Wouldn't we want to keep libc6Cmd as a member of the Metrics struct so it can be mocked the same way as the other commands?

Reply:
I think we want to only mock it on the Metrics constructor, don’t we? We don’t necesserally need to store it on the struct once used.

Reply:
Okay I see the point, you'd prefer to keep the Metrics struct as small as possible. This will result in a few more local variables being created and commands being executed in the constructor since we'll need to get the architecture and libc6 version in the constructor in order to directly assign m.hwCapCmd

Reply:
That’s fine with me :)